docs/DeveloperGuide.md

```diff
@@ -520,23 +520,31 @@ Use case ends.
-8. As the app is an offline tool, it should be robust and able to run for extended periods without crashing. Downtime should be limited to under 1% (if any issues require restarting).
-9. The system should be able to recover from crashes within 5 seconds, ensuring minimal disruption to the user’s workflow.
-10. Since celebrity data is highly sensitive, the application must store all information securely (e.g. locally) and ensure no unauthorized access can occur, even in an offline environment.
-11. The app should allow password protection and encryption options to ensure that only authorized users can access or modify sensitive data.
-12. The system should log all critical operations (e.g., adding or deleting contacts) for auditing purposes, ensuring traceability of actions.
-13. The system should provide a way to export and import data in a standard format (e.g., CSV) to allow for easy backups and data transfer between devices.
-14. The system should provide a way to archive old data to reduce clutter and improve performance, ensuring that archived data can be easily retrieved if needed.
-15. New commands or features should be easily added or modified without impacting the existing system. This includes the ability to update or patch the app efficiently.
-16. Since the app is intended to be an offline tool, it should function fully without any internet connection. All features, including scheduling and contact management, must be operable offline.
-17. The application should be lightweight, with minimal CPU and memory usage, allowing it to run smoothly on a wide variety of machines, including low-spec systems.
-18. The system should provide a user-friendly and intuitive interface with clear instructions and feedback to guide users who are not IT-savvy in using the command-line interface effectively.
+2. The software should work without requiring an installer.
+3. The software should not depend on your own remote server. 
+4. Since the app is intended to be an offline tool, it should function fully without any internet connection.
+5. The application should be packaged into a single JAR file.
+6. The file sizes of the deliverables should be reasonable and not exceed the limits given below. 
+   - Product (i.e., the JAR/ZIP file): 100MB
+   - Documents (i.e., PDF files): 15MB/file
+7. The final JAR/PDF files cannot be bloated unnecessarily.
+8. The developer guide and user guide should be PDF-friendly. Don't use expandable panels, embedded videos, animated GIFs etc.
+9. The use of third-party libraries/frameworks/services are not allowed unless they are approved by the teaching team.
+10. The software should follow the Object-oriented paradigm primarily.
+11. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
```

Comment:
Phrase it into something more related to our product? "Our software should enable talent managers with above average..."